pysynchronizer/cache.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2021 Daniel P. Smith, Apertus Solutions LLC
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+from .singleton import Singleton
+
+class Cache(Singleton):
+    def __init__(self, namespaces):
+        self.internal_store = {}
+        for namespace in namespaces:
+            self.internal_store[namespace] = {}
+
+    def set_key(self, namespace, name, value):
+        self.internal_store[namespace][name] = value
+
+    def append_key(self, namespace, name, value):
+        self.internal_store[namespace][name].update(value)
```

Comment:
I implemented the cache initially so results could be cached in interactive mode, but in testing with and without caching, there was no noticeable impact on performance. And at this point, our internal use case will likely rarely if ever use sync-cmd interactively. I left the cache class here in case there was a desire to have it, but I can drop it for now.